src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerPerTreeOptionsProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal sealed partial class CompilerPerTreeOptionsProvider : PerTreeOptionsProvider
+    {
+        public static OptionSet EmptyCompilerOptionSet { get; } = new CompilerOptionSet();
```

Comment:
Why use OptionSet here instead of CompilerOptionSet? This is an internal API, no need to hide the type here.